src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -83,13 +85,22 @@ void initializeReference() {
      * Package-private so that engine classes can access it, but concrete tool child classes cannot.
      * May be overridden by traversals that require custom initialization of the reads data source.
      */
-    void initializeReads() {
+    void initializeReads() throws UserException {
         SamReaderFactory factory = null;
         if (hasReference()){
             // pass in reference if available, because CRAM files need it
             factory = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT).referenceSequence(referenceArguments.getReferenceFile());
         }
         reads = ! readArguments.getReadFiles().isEmpty() ? new ReadsDataSource(readArguments.getReadFiles(), factory) : null;
+        boolean hasCram = false;
+        for (File inputFile : readArguments.getReadFiles()) {
+            if (FilenameUtils.getExtension(inputFile.getName()).equals("cram")) {
+                hasCram = true;
+            }
+        }
```

Comment:
Extract this loop into a private utility method `hasCramInput()`, and call that method from `initializeReads()`